crowbar_framework/spec/lib/crowbar/lock/shared_non_blocking_spec.rb

```diff
@@ -0,0 +1,185 @@
+require "spec_helper"
+require "timeout"
+require "rspec/expectations"
+
+RSpec::Matchers.define :be_unlocked do
+  match do |actual|
+    !actual.locked? && !File.exists?(actual.path)
+  end
+  failure_message do |actual|
+    if actual.locked?
+      "expected that #{actual} would not be locked"
+    else
+      "expected that #{actual.path} would not exist"
+    end
+  end
+end
+
+RSpec::Matchers.define :be_locked_by_me do
+  match do |actual|
+    actual.locked? && File.exists?(actual.path)
+  end
+  failure_message do |actual|
+    if !actual.locked?
+      "expected that #{actual} would be locked"
+    else
+      "expected that #{actual.path} would exist"
+    end
+  end
+end
+
+RSpec::Matchers.define :be_locked_by_another do
+  match do |actual|
+    !actual.locked? && File.exists?(actual.path)
+  end
+  failure_message do |actual|
+    if actual.locked?
+      "expected that #{actual} would not be locked"
+    else
+      "expected that #{actual.path} would exist"
+    end
+  end
+end
+
+describe Crowbar::Lock::SharedNonBlocking do
+  def safe_pipe(cmd, stdin)
+    IO.popen(cmd, 'w+') do |pipe|
```

Comment:
Style/StringLiterals: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.